pjsip/src/pjsua2/media.cpp

```diff
@@ -265,6 +265,78 @@ AudioMedia* AudioMedia::typecastFromMedia(Media *media)
 
 ///////////////////////////////////////////////////////////////////////////////
 
+AudioMediaPort::AudioMediaPort()
+: pool(NULL)
+{
+}
+
+AudioMediaPort::~AudioMediaPort()
+{
+    if (pool) {
+        PJSUA2_CATCH_IGNORE( unregisterMediaPort() );
+        pj_pool_release(pool);
+        pool = NULL;
+    }
+}
+
+static pj_status_t get_frame(pjmedia_port *port, pjmedia_frame *frame)
+{
+    AudioMediaPort *mport = (AudioMediaPort *) port->port_data.pdata;
+    MediaFrame frame_;
+
+    frame_.size = frame->size;
+    mport->getFrame(frame_);
+    frame->type = frame_.type;
+    frame->size = PJ_MIN(frame_.buf.size(), frame_.size);
+    pj_memcpy(frame->buf, frame_.buf.c_str(), frame->size);
```

Comment:
According to the doc:
https://cplusplus.com/reference/string/string/c_str/
it should be safe, since the returned array must `include the same sequence of characters that make up the value of the string object`, so there won't be truncation.

```
    const char byteArray[] = {0x48, 0x65, 0x6C, 0x6C, 0x6F, 0x00, 0x77, 0x6F, 0x72, 0x6C, 0x64};
    std::string byteString(byteArray, sizeof(byteArray));

    std::cout << "Byte String: " << byteString << std::endl;
    std::cout << "Length: " << byteString.length() << std::endl;

    std::cout << "Byte String: " << byteString.c_str() << std::endl;

    char buf[sizeof(byteArray)];
    pj_memcpy(buf, byteString.c_str(), byteString.length());
    std::cout << "c_str copy, buf == byteArray? " << (pj_memcmp(buf, byteArray, sizeof(byteArray))? "different": "same") << std::endl;

    byteString.copy(buf, byteString.length());
    std::cout << "string copy, buf == byteArray? " << (pj_memcmp(buf, byteArray, sizeof(byteArray))? "different": "same") << std::endl;

    std::copy(byteString.begin(), byteString.end(), buf);
    std::cout << "std copy, buf == byteArray? " << (pj_memcmp(buf, byteArray, sizeof(byteArray))? "different": "same") << std::endl;
```

Output:
```
Byte String: Helloworld
Length: 11
Byte String: Hello
c_str copy, buf == byteArray? same
string copy, buf == byteArray? same
std copy, buf == byteArray? same
```

After comparing the three options, now I prefer std::string copy
https://cplusplus.com/reference/string/string/copy/
`Copies a substring of the current value of the string object into the array pointed by s. This substring contains the len characters that start at position pos. The function does not append a null character at the end of the copied content.`

What do you think?

Reply:
Sounds good if the spec says the content won't be truncated. Also, we need to be careful as C++ specs may be changed (slightly) among versions.

Reply:
In the latest commit, change the buffer to use `std::vector` as suggested.
Also add a usage example in Python.

Note that threading must be enabled. Should we leave this as default? Not sure why anyone would want it disabled nowadays.